transaction/src/action/output.rs

```diff
@@ -22,6 +25,45 @@ impl IsAction for Output {
     fn balance_commitment(&self) -> balance::Commitment {
         self.body.balance_commitment
     }
+
+    fn decrypt_with_perspective(
+        &self,
+        txp: &TransactionPerspective,
+    ) -> anyhow::Result<Option<ActionView>> {
+        // Get payload key for note commitment of note payload
+
+        let note_commitment = self.body.note_payload.note_commitment;
+
+        // Get payload key for note commitment of swap NFT.
+        let payload_key = txp
+            .payload_keys
+            .get(&note_commitment)
+            .ok_or_else(|| anyhow::anyhow!("corresponding payload key not found"))?;
```

Comment:
This error message is pretty opaque, and I think it won't give much information if it's encountered. Perhaps
```
perspective is missing payload key for output note commitment {}
```
would be better, since it identifies exactly what's missing and in what context.